apps/ledger-live-mobile/src/newArch/features/Accounts/screens/AddAccount/components/SelectAddAccountMethod/useSelectAddAccountMethodViewModel.ts

```diff
@@ -29,10 +29,15 @@ const useSelectAddAccountMethodViewModel = ({
   const navigationParams = useMemo(() => {
     return hasCurrency
       ? currency.type === "TokenCurrency"
-        ? { token: currency }
-        : { currency }
-      : {};
-  }, [hasCurrency, currency]);
+        ? {
+            token: currency,
+            ...(llmNetworkBasedAddAccountFlow?.enabled && { context: "addAccounts" }),
+          }
+        : { currency, ...(llmNetworkBasedAddAccountFlow?.enabled && { context: "addAccounts" }) }
+      : llmNetworkBasedAddAccountFlow?.enabled
+        ? { context: "addAccounts" }
+        : {};
+  }, [hasCurrency, currency, llmNetworkBasedAddAccountFlow?.enabled]);
```

Comment:
Could we please improve readability? 

Maybe use `if else` instead of ternary ?